### PR TITLE
[FEATURE] Build Hero Section Full #5

### PR DESCRIPTION
**Pull request by Vansh-Baghel**
* Completed the HeroSection as per figma design.
* Added text, with accordance to the fonts mentioned.
* Added incrementing numbers animation in counts section.
* Added carousel for each image (Images ain't decided, so have kept the same for 4 slides).
* Made completely responsive.

### CODE REVIEW

File: src/components/Hero/HeroSection.astro [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+import CountUp from "react-countup";
+import GridWrapper from "./GridWrapper";
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: remove this. use tailwind grid classes. 
This is not required. same thing can be made using normal tailwind.

File: src/components/Hero/HeroSection.astro [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      >
+        The Training and Placement department serves as a bridge between the
+        academic environment and the industry by establishing connections with
+        various companies and organizations. The ultimate aim of the T&P
+        department is to ensure that students are well-prepared for the job
+        market and have access to the resources they need to achieve their
+        career goals.
+      </div>
+      <button
+        class="my-6 p-3 rounded-2xl m-auto"
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: keep it left aligned

File: src/components/Hero/HeroSection.astro [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        style={{ border: "2px solid #E5E5E5" }}
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: use tailwind border classes here. All styling is possible through styling. 
Use this for systematic follow up.

Do this everywhere u used style attribute

File: src/components/Hero/HeroSection.astro [hunk 4]
```diff
@@ -0,0 +1,107 @@
+---
+import CountUp from "react-countup";
+import GridWrapper from "./GridWrapper";
+import HeroSwiper from "./HeroSwiper";
+---
+
+<>
+  {/* Text of the HeroSection */}
+  <div class="flex max-md:grid">
+    <div class="p-10 w-[50%] max-md:w-[100%]">
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: we are doing mobile first. and u are doing it in reverse
it should be grid md:flex 
not the other way around

File: src/components/Hero/HeroSection.astro [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        class="my-6 p-3 rounded-2xl m-auto"
+        style={{ border: "2px solid #E5E5E5" }}
+      >
+        View More
+      </button>
+    </div>
+  </div>
+
+  {/* Numbers and text inside it */}
+  <div class="flex justify-evenly max-lg:grid">
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: same, do this mobile first

File: src/components/Hero/HeroSection.astro [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    <div class="ml-6 max-w-[40%] max-lg:max-w-[100%]">
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: same do this mobile first

File: src/components/Hero/HeroSection.astro [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      <GridWrapper client:load columns={2}>
+        <div
+          class="max-w-[180px] p-4 m-3 rounded-2xl"
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: don't use pixels. use rems. or default tailwind values max-w-16 something will be around 180px. hover over that utility in vscode to know the exact value

File: src/components/Hero/HeroSection.astro [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+          style={{ border: "2px solid #E5E5E5" }}
+        >
+          {" "}
+          <div class="text-title text-2xl">
+            <CountUp start={0} end={852} />+{" "}
+          </div>
+          <div class="">Number of students placed</div>
+        </div>
+        <div
+          class="max-w-[180px] p-4 m-3 rounded-2xl"
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: same, use rems not pixels

File: src/components/Hero/HeroSection.astro [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+          {" "}
+          <div class="text-title text-2xl">
+            <CountUp start={0} end={7400} />+{" "}
+          </div>
+          <div class="">Number of companies visited</div>
+        </div>
+      </GridWrapper>
+      <GridWrapper client:load columns={2}>
+        <div
+          class="max-w-[180px] p-4 m-3 rounded-2xl"
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: same, use rems not pixels

File: src/components/Hero/HeroSection.astro [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+          style={{ border: "2px solid #E5E5E5" }}
+        >
+          {" "}
+          <div class="text-title text-2xl">
+            <CountUp start={0} end={7400} />+{" "}
+          </div>
+          <div class="">Students with 10+ LPA packages</div>
+        </div>
+        <div
+          class="max-w-[180px] p-4 m-3 rounded-2xl"
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: same, use rems not pixels

**Review by JayeshVP24 — CHANGES_REQUESTED** [review]

File: src/components/Hero/HeroSection.jsx [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+          <rect width="24" height="24" rx="12" fill="#E4E7EC"></rect>
+          <path
+            d="M6.16663 12H17.8333M17.8333 12L12 6.16669M17.8333 12L12 17.8334"
+            stroke="#101828"
+            stroke-width="1.5"
+            stroke-linecap="round"
+            stroke-linejoin="round"></path>
+        </svg>
+      </button>
+      <div className="bold font-title text-[2.4rem] max-lg:text-[2rem]">
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.jsx:31]: this bold doesn't work here. remove it. not needed

File: src/components/Hero/HeroSection.astro [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+  <div class="flex justify-evenly max-lg:grid">
+    <div class="ml-6 max-w-[40%] max-lg:max-w-[100%]">
+      <GridWrapper client:load columns={2}>
+        <div
+          class="max-w-[180px] p-4 m-3 rounded-2xl"
+          style={{ border: "2px solid #E5E5E5" }}
+        >
+          {" "}
+          <div class="text-title text-2xl">
+            <CountUp start={0} end={852} />+{" "}
```
Comment (JayeshVP24) [inline src/components/Hero/HeroSection.astro]: add client:load directive here. it will work. do this for all countups

**Review by JayeshVP24 — CHANGES_REQUESTED** [review]

File: src/components/Hero/HeroSection.astro (hunk 12 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: I got to create a new jsx for countup, as it wasn't working directly by importing.

File: src/components/Hero/HeroSection.jsx (hunk 11 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.jsx:31]: done

File: src/components/Hero/HeroSection.astro (hunk 10 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 9 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 8 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 7 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 6 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 5 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 4 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 3 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 2 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 1 above)
Comment (Vansh-Baghel) [reply src/components/Hero/HeroSection.astro]: done

File: src/components/Hero/HeroSection.astro (hunk 12 above)
Comment (JayeshVP24) [reply src/components/Hero/HeroSection.astro]: yes, that is the right way to do it

**Review by JayeshVP24 — APPROVED** [review]
LGTM

**Pull request merged**